### PR TITLE
Audit fixes

### PR DESCRIPTION
**Pull request by juztamau5**
## Description

This PR addresses issues from the Bramah Systems audit.

One issue was fixed earlier in https://github.com/wolvesofwallstreet/wolves.finance/commit/0efbb0ac55dad8d5c6e08162c02ab68523594923. Most of the remaining issues are fixed in the commits here.

One item not addressed is:

```
## Unused variables should be removed
Controller.sol references an “amount” variable that should be removed.
```

@peak3d can you decide what do you about this one?

### CODE REVIEW

File: contracts/src/crowdsale/WOWSSftMinter.sol [hunk 1]
```diff
@@ -41,7 +41,7 @@ contract WOWSSftMinter is Context, Ownable {
   }
   mapping(uint256 => CFolioItemSft) public cfolioItemSfts; // C-folio type to c-folio data
 
-  uint256 public nextCFolioItemNft = 0x10000000000000000;
+  uint256 public nextCFolioItemNft = (1 << 64);
```
Comment (peak3d) [inline contracts/src/crowdsale/WOWSSftMinter.sol:44]: Have you verified that the 1 is treated as uint256 ? I had issues with that in the past without typecasting like uint256(1)<<64

File: contracts/src/utils/TokenIds.sol [hunk 2]
```diff
@@ -13,17 +13,16 @@ library TokenIds {
   uint256 public constant HASH_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
 
   function isBaseCard(uint256 tokenId) internal pure returns (bool) {
-    return (tokenId & HASH_MASK) < 0x10000000000000000;
+    return (tokenId & HASH_MASK) < (1 << 64);
```
Comment (peak3d) [inline contracts/src/utils/TokenIds.sol:16]: same, IMO the 1 has to be casted to an uint256(1)

File: contracts/src/utils/TokenIds.sol [hunk 3]
```diff
@@ -13,17 +13,16 @@ library TokenIds {
...
   }
 
   function isStockCard(uint256 tokenId) internal pure returns (bool) {
-    return (tokenId & HASH_MASK) < 0x100000000;
+    return (tokenId & HASH_MASK) < (1 << 32);
```
Comment (peak3d) [inline contracts/src/utils/TokenIds.sol:20]: same

**Review by peak3d — COMMENTED** [review]
Thx for fixing, but we need to be sure that this is still working.
Remember the shift bug in ERC1155.

**Comment by peak3d** [comment]
Regarding the unused amount: sure, let us remove it from impl.

```
--- a/contracts/src/investment/Controller.sol
+++ b/contracts/src/investment/Controller.sol
@@ -123,7 +123,7 @@ contract Controller is IController, Context, Ownable {
   /**
    * @dev See {IController-onDeposit}
    */
-  function onDeposit(uint256 amount)
+  function onDeposit(uint256/* amount*/)
     external
     view
     override
@@ -137,16 +137,13 @@ contract Controller is IController, Context, Ownable {
     require(farm.farmEndedAtBlock == 0, 'Farm closed');
     require(!farm.paused, 'Farm paused');

-    // Unused
-    amount;
-
     return 0;
   }

   /**
    * @dev See {IController-onDeposit}
    */
-  function onWithdraw(uint256 amount)
+  function onWithdraw(uint256/* amount*/)
     external
     view
     override
@@ -155,9 +152,6 @@ contract Controller is IController, Context, Ownable {
     // Validate state
     require(!farms[_msgSender()].paused, 'Farm paused');

-    // Unused
-    amount;
-
     return 0;
   }

```

File: contracts/src/utils/TokenIds.sol [hunk 4]
```diff
@@ -13,17 +13,16 @@ library TokenIds {
   uint256 public constant HASH_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```
Comment (peak3d) [inline contracts/src/utils/TokenIds.sol]: (1 << 128) -1

**Review by peak3d — CHANGES_REQUESTED** [review]
I tested the (1 << XX) things and they work as expected.
can you finally please change the hashmask? Then it can go in

**Comment by juztamau5** [comment]
Thanks for testing. I added the hashmask. CI isn't green but I verified that it passes when based onto the last success so this is good to go.

**Pull request merged**